_posts/2018-07-25-strimzi-0-5-0-release.md

```diff
@@ -0,0 +1,166 @@
+  name: my-cluster
+spec:
+  kafka:
+    replicas: 3
+    config:
+      offsets.topic.replication.factor: 3
+      transaction.state.log.replication.factor: 3
+      transaction.state.log.min.isr: 2
+    storage:
+      type: ephemeral
+...
+...
+zookeeper:
+    replicas: 3
+...
+```
+
+Kafka is now a first-class citizen in the Kubernetes/OpenShift world and the cluster deployment can be managed through the `kubectl` and `oc` tools natively.
+For example, you can get Kafka cluster information through the following command:
+
+```
+oc describe kafka my-cluster
+```
+
+So `kafka` is now at same level as any other Kubernetes/OpenShift native resourse, for example `configmap`, `deployment`, `secret` and so on.
+The same is true for Kafka Connect cluster which is defined by a `KafkaConnect` or `KafkaConnectS2I` Custom Resources.
+
+# Encryption support
+
+The communication across the deployed Kafka cluster now uses the TLS protocol and thus benefits from both encryption and mutual authentication.
+More precisely, the following traffic is now encrypted:
+
+* Zookeeper cluster communication
+* Kafka cluster commbunication
+* Communication between Kafka and Zookeeper
+* Communication between the Topic Operator and Kafka / Zookeeper
+
+Setting up the TLS infrastructure is done by the Cluster Operator which generates the certificates needed for encryption.
+At same time a clients CA certificate is provided within a Secret so that the user can use it for signing clients' certificates and allowing client authentication for those clients which want to connect via TLS.
+
+# Logging configuration improvements
+
+In order to improve the logging experience, it's now possible to specify the logging level (i.e. `DEBUG`, `INFO`, ...) in the Custom Resource describing the cluster both for Kafka and Zookeeper.
+There are two different ways of doing this, as showed by the following example.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    ...
+    logging:
+      type: inline
+      loggers:
+        kafka.root.logger: "INFO,CONSOLE"
+    ...
+  zookeeper:
+    ...
+    logging:
+      type: external
+      name: customConfigMap
+    ...
+```
+
+The first specifies the log levels inline (that is, directly in the Kafka resource).
+
+The other way is to provide a reference to a complete log4j configuration stored externally in a ConfigMap.
+
+# Rack awareness
+
+In order to have high-availability, Apache Kafka provides a feature called "rack awareness".
+When your topics are replicated across multiple brokers, it is desirable for those brokers to be in different physical racks.
+If a rack fails, you are sure that other topic replicas are alive, a new leader can be elected and clients can continue to exchange messages. As opposite, if the brokers hosting replicas for a topic are put into the same rack, the partitions are not available anymore on rack failure because all brokers will go down at same time.
```

Comment:
This is because replica assignment that is unaware of broker racks can result in partitions being assigned to brokers in the same rack. Failure of that rack would thus make multiple replicas unavailable, defeating the point of having replicas.